array.c

```diff
@@ -1512,35 +1512,38 @@ rb_ary_shift(VALUE ary)
 
 /*
  *  call-seq:
- *     array.shift -> object or nil
- *     array.shift(n) -> new_array
+ *    shift -> new_array or nil
+ *    shift(count) -> new_array or nil
  *
- *  Removes and returns leading elements.
+ *  Removes and returns leading elements from +self+ in a new array.
```

Comment:
It only returns a new array when `count` is given.

Reply:
Fixed.